src/tax-lot/tax-lot.repository.ts

```diff
@@ -70,6 +77,100 @@ export class TaxLotRepository {
     }
   }
 
+  async findGeomFromGeoJson(
+    shape: Geometry,
+    targetSrid = 4326,
+  ): Promise<FindGeomFromGeoJsonRepo> {
+    try {
+      const result = await this.db.execute(
+        sql`SELECT ST_Transform(ST_GeomFromGeoJSON(${shape}), CAST(${targetSrid} AS INTEGER)) AS geom`,
+      );
+      return result.rows[0] as { geom: Geom };
+    } catch {
+      throw new DataRetrievalException();
+    }
+  }
+
+  async findGeomBuffer(
+    geom: Geom,
+    buffer: number,
+  ): Promise<FindGeomBufferRepo> {
+    try {
+      const result = await this.db.execute(
+        sql`SELECT ST_Buffer(${geom}, ${buffer}) AS buffer`,
+      );
+      return result.rows[0] as { buffer: Geom };
+    } catch {
+      throw new DataRetrievalException();
+    }
+  }
+
+  async checkGeomIsValid(geom: Geom): Promise<CheckGeomIsValidRepo> {
```

Comment:
Could the return type of this just be `boolean`? I know you're using a type infered from a zod schema but I think you could have a zod schema that is an instance of `z.boolean()` instead of an object containing an instance of `z.boolean()`.